### ISSUE TITLE
Can't drag audio files

### ISSUE THREAD

**Issue by pejrich** [publicsamples/Oi-Grandad#14]
If I try to drag an audio file in Ableton it just acts as if the plugin is not there and as if I'm dragging into a session clip. The only way to work seems to be using the right clck, but then it'll only let me add .wav and I have to navigate each time. Is there no way to drag directly from ableton browser to the plugin?

**Comment by modularsamples** [comment]
Drag and drop only works from your OS file browser, finder and so on. It's not directly integrated with the DAW. This would actually be a really cool feature, but I'm not aware of any 3rd party plugin that does this. for example, we can't drag and drop into Kontakt the same way we could with a native plugin, like Sampler

I'll look into it, but don't hold your breath..

**Comment by pejrich** [comment]
@modularsamples That's true, I may not have seen it on other plugins, but what I have seen is the ability to drag your file from the ableton browser onto the plugin frame in ableton in the devices section, but I'm not sure how that would work considering this plugin has potentially multiple(looked like 4 max) places the file could be for. Unless there's a way to detect a drag/drop and display a quadrant. 

I would be happy to take a look at it myself, though I've never worked with VST or plugins before, taking a quick look at the code I see empty JS functions, so I may need a pointer in the right direction.

**Comment by modularsamples** [comment]
Well, it certainly looks possible. 

First off, you should know that Oi, grandad is built with [HISE](http://hise.audio), HISE is built with [JUCE](https://juce.com). 

See this post on the JUCE forums (Assuming that if you can drag a track, you can drag a file??):
https://forum.juce.com/t/drag-and-drop-tracks/50698/3

As for implementing this in HISE, IDK. There's not much information on it. I'd start here:

**HISE Scripting API**
https://docs.hise.audio/scripting/scripting-api/index.html

**HISE post about Dragging midi files TO a DAW**
https://forum.hise.audio/topic/4575/midi-player-and-drag-to-daw-almost-there-speed-buttons-success?_=1669384038705

If this is something you'd like to look into, fork the project and I'll assign this issue to you.
